provider/provider.go

```diff
@@ -113,3 +113,48 @@ func pinSet(ctx context.Context, pinning pin.Pinner, fetchConfig fetcher.Factory
 
 	return set, nil
 }
+
+func NewPrioritizedProvider(streams ...KeyChanFunc) KeyChanFunc {
+	return func(ctx context.Context) (<-chan cid.Cid, error) {
+		outCh := make(chan cid.Cid)
+
+		go func() {
+			defer close(outCh)
+			has := map[string]bool{}
+
+			for _, stream := range streams {
+				ch, err := stream(ctx)
+				if err != nil {
+					log.Warnf("error in prioritized strategy: %w", err)
+				} else {
+				L:
```

Comment:
Adding the `continue` here is not going to help removing the `L` since the problem is the that you need to break out of the nested `select` to the `for`. There's no other way of doing this here.

Anyways, it does not matter since this is going to be reworked to only work with two streams.